src/main/java/htsjdk/samtools/util/OverlapDetector.java

```diff
@@ -82,13 +82,19 @@ public void addLhs(final T object, final Locatable interval) {
         final int start = interval.getStart() + this.lhsBuffer;
         final int end   = interval.getEnd()   - this.lhsBuffer;
 
-        final Set<T> objects = new HashSet<>(1);
-        objects.add(object);
+        final Set<T> newValue = Collections.singleton(object);
         if (start <= end) {  // Don't put in sequences that have no overlappable bases
-            final Set<T> alreadyThere = tree.put(start, end, objects);
+            final Set<T> alreadyThere = tree.put(start, end, newValue);
             if (alreadyThere != null) {
-                alreadyThere.add(object);
-                tree.put(start, end, alreadyThere);
+                if( alreadyThere.size() == 1){
```

Comment:
I think that this should be implemented within the IntervalTree class as a `computeIfPresent` method...